src/components/include/utils/logger.h

```diff
@@ -98,55 +98,86 @@ class Logger {
   static void SetLogger(Pimpl& impl);
   static Pimpl& GetLogger();
 
+  static void SetLoggerPtr(LoggerType& logger) {
+    logger_ = logger;
+  }
+
+  static LoggerType& GetLoggerPtr() {
+    return logger_;
+  }
+
  private:
   Logger();
   Logger(const Logger&);
   Logger& operator=(const Logger&);
   ~Logger();
 
   static Pimpl impl_;
+  static LoggerType logger_;
+};
+
+class Log {
+ public:
+  Log(const bool logs_enabled) {
+#if defined(LOG4CXX_LOGGER)
+
+    logger::Logger::InitLogger(logs_enabled, "log4cxx.properties");
+#else
+    logger::Logger::InitLogger(
+        logs_enabled, logger::LogLevel::LL_TRACE, "SmartDeviceLink.log");
```

Comment:
What do you mean? I didn't changed logic init logger.
`#undef INIT_LOGGER
#if defined(LOG4CXX_LOGGER)
#define INIT_LOGGER(logs_enabled)             \
  if (logs_enabled) {                         \
    DCHECK(utils::appenders_loader.Loaded()); \
  }                                           \
  logger::Logger::InitLogger(logs_enabled, "log4cxx.properties");
#else
#define INIT_LOGGER(logs_enabled) \
  logger::Logger::InitLogger(     \
      logs_enabled, logger::LogLevel::LL_TRACE, "SmartDeviceLink.log");
#endif`